### PR TITLE
add to the Nginx HTTPS section

### PR DESCRIPTION
**Pull request by tkimnguyen**

### CODE REVIEW

File: manage/deploying/front-end/nginx.rst [hunk 1]
```diff
@@ -539,12 +539,97 @@ More info
...
+	ssl_certificate_key /etc/ssl/localcerts/mydomain.com.key;
+	server_name mydomain.com;
+	access_log /var/log/nginx/mydomain.com.access.log;
+	error_log /var/log/nginx/mydomain.com.error.log;
+
+	# Note that domain name spelling in VirtualHostBase URL matters
+	# -> this is what Plone sees as the "real" HTTP request URL.
+	# "Plone" in the URL is your site id (case sensitive)
+	location / {
+	    rewrite ^/(.*)$ /VirtualHostBase/https/mydomain.com:443/MyDomain/VirtualHostRoot/$1 break;
```
Comment (polyester) [inline manage/deploying/front-end/nginx.rst]: shouldn't "MyDomain" here be replaced with "Plone" ??

Comment (cleberjsantos) [inline manage/deploying/front-end/nginx.rst]: @polyester +1 And from /VirtualHostBase/**https/** to /VirtualHostBase/**$scheme/**?

Comment (tkimnguyen) [inline manage/deploying/front-end/nginx.rst]: @polyester ah yeah I could add a note of explanation about that. I'm used to running multiple Plone sites in a given Zope so to default to "Plone" is confusing to _me_ (seems like some implicit magic keyword, as opposed to having it explained as "the ID of your Plone site").

@cleberjsantos I did not know about that $scheme variable - thank you.

**Comment by svx** [comment]
Maybe we should update the ssl part to something what is a bit more and not just the very basics, like we should add ciphers, http2, ocsp stapling, Forward Secrecy & Diffie Hellman Ephemeral Parameters ?
Also tls versions could be added, there is lots of more stuff like Logjam, HTTP Strict Transport Security and HTTP Public Key Pinning Extension.

**Comment by gforcada** [comment]
@svx isn't that better suited to send to an external website? Chipers and so on get outdated every now and then, so if there is a www.configure-ssl-the-good-way.com I would redirect there...

**Comment by svx** [comment]
Yes I would do that too, but IMHO we should mention that with useful links and so on

**Comment by cleberjsantos** [comment]
I use the conf as below

```
server {
   listen 443 ssl default_server;
   server_name mydomain.com;

   access_log /var/log/nginx/mydomain.com.access.log;
   error_log /var/log/nginx/mydomain.com.error.log;

   if ($host !~* "mydomain.com") {
       rewrite ^(.*)$ https://mydomain.com$1 permanent;
       break;
   }
   ssl on;
   ssl_certificate           /etc/ssl/localcerts/mydomain.com.crt;
   ssl_certificate_key   /etc/ssl/localcerts/mydomain.com.key;
   ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
   ssl_ciphers "ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-SHA384:ECDHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-SHA:ECDHE-RSA-AES128-SHA:DHE-RSA-AES256-SHA256:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:ECDHE-RSA-DES-CBC3-SHA:EDH-RSA-DES-CBC3-SHA:AES256-GCM-SHA384:AES128-GCM-SHA256:AES256-SHA256:AES128-SHA256:AES256-SHA:AES128-SHA:DES-CBC3-SHA:HIGH:!aNULL:!eNULL:!EXPORT:!DES:!MD5:!PSK:!RC4";

   ssl_prefer_server_ciphers on;
   add_header Strict-Transport-Security "max-age=31536000; includeSubDomains";


  # Note that domain name spelling in VirtualHostBase URL matters
  #-> this is what Plone sees as the "real" HTTP request URL.
  # "Plone" in the URL is your site id (case sensitive)
  location / {
     rewrite ^/(.*)$ /VirtualHostBase/$scheme/mydomain.com:443/Plone/VirtualHostRoot/$1 break;
     proxy_http_version 1.1;
     proxy_set_header Connection "";
     proxy_ignore_headers Expires;
     proxy_set_header X-Real-IP $remote_addr;
     proxy_set_header Host $http_host;
     proxy_pass_header Set-Cookie;
     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
     proxy_set_header X-Forwarded-Proto https;
   }
}
```

With **spdy**:

```
...
  listen 443 ssl spdy default_server;
...
   add_header Alternate-Protocol  443:npn-spdy/3;
   spdy_headers_comp       6;
   if ($http_cookie ~* "__ac=([^;]+)(?:;|$)" ) {
       break;
   }
```

Enable **HTTP/2**:

If you’re already running SPDY, please note that the SPDY module have been replaced with the HTTP/2 module in Nginx. Fortunately, to run HTTP/2 you just need to upgrade to Nginx 1.9.5 or later and replace spdy with http2 on your listen directive line, but, before, make sure your version of Nginx is compiled with HTTP/2 support:

```
$ nginx -V
```

Now it is as simple as adding a single word to your Nginx config:

```
...
   listen 443 ssl http2 default_server;
...
```

**Comment by tkimnguyen** [comment]
I've made the changes suggested by @polyester and @cleberjsantos. Let's get this merged and if someone wants to add a lot more to the section then that's great!

File: manage/deploying/front-end/nginx.rst (hunk 1 above)
Comment (tkimnguyen) [inline manage/deploying/front-end/nginx.rst]: Changes made

File: manage/deploying/front-end/nginx.rst [hunk 2]
```diff
@@ -539,12 +539,99 @@ More info
...
+	ssl_certificate_key /etc/ssl/localcerts/mydomain.com.key;
+	server_name mydomain.com;
+	access_log /var/log/nginx/mydomain.com.access.log;
+	error_log /var/log/nginx/mydomain.com.error.log;
+
+	# Note that domain name spelling in VirtualHostBase URL matters
+	# -> this is what Plone sees as the "real" HTTP request URL.
+	# "Plone" in the URL is your site id (case sensitive)
+	location / {
+	    rewrite ^/(.*)$ /VirtualHostBase/$scheme/mydomain.com:443/MyDomain/VirtualHostRoot/$1 break;
```
Comment (polyester) [inline manage/deploying/front-end/nginx.rst]: still s/MyDomain/Plone needed (or change it to MySiteId and change line 616 accordingly, if you want to point out that "Plone" here is not magical, but simply a Site ID.)

**Comment by svx** [comment]
Hum I just checked our docs on nginx and port 80, and there we use

```
upstream plone {
    server 127.0.0.1:8090;
}
```

and

```
 location / {
          proxy_pass http://plone/VirtualHostBase/http/yoursite.com:80/Plone/VirtualHostRoot/;
    }
}
```

I suggest we use the same example on both 80 and 443 otherwise it is confusing, so we should go with one or the other, but not with two different examples :)

**Comment by tkimnguyen** [comment]
@svx ah yeah I'd noticed that 8090 port before - I just thought it was a typo. I think it should be changed to 8080. 

OK I'll go to the Plone as site ID and and "yoursite.com" as the domain.

Man you guys are _killing_ me ... when the only things being killed today should be the 280 million turkeys! :)

**Comment by tkimnguyen** [comment]
All right: are we ok now? :) :+1:

**Comment by svx** [comment]
Nope :)

it was not about the port number, well 8080 is of course ok too, it is about:

about the usage of proxy_pass, which uses upstream plone, you can see that here:

```
# This specifies which IP and port Plone is running on.
# The default is 127.0.0.1:8080
upstream plone {
    server 127.0.0.1:8090;
}

# Redirect all www-less traffic to the www.site.com domain
# (you could also do the opposite www -> non-www domain)
server {
    listen 80;
    server_name yoursite.com;
    rewrite ^/(.*) http://www.yoursite.com/$1 permanent;
}

server {

    listen 80;
    server_name www.yoursite.com;
    access_log /var/log/nginx/yoursite.com.access.log;
    error_log /var/log/nginx/yoursite.com.error.log;

    # Note that domain name spelling in VirtualHostBase URL matters
    # -> this is what Plone sees as the "real" HTTP request URL.
    # "Plone" in the URL is your site id (case sensitive)
    location / {
          proxy_pass http://plone/VirtualHostBase/http/yoursite.com:80/Plone/VirtualHostRoot/;
    }
}
```

You use `rewrite` in the location part, meaning we have to different ways in the docs, and we should decide on one, otherwise it is confusing

**Comment by tkimnguyen** [comment]
I don't have an opinion on which one to use. Could you merge this then make your suggested change or tell me what to put in? I really only wanted to help and I pulled together this working example from other docs I found.

**Comment by svx** [comment]
I am really thankfully for what you are doing ! 
But, yes sorry for the but :) I do not like to merge something, when we already know that it is not there yet, and we have then branch it again to finish/fix it.

This will also decrease the docs quality temp.

So what is with the following workaround:
- Delete your current PR
- You create a branch in this repo here and not your own one
- We will fix the branch, since now people can work on it
- Merge it into master, when we are done 
- Have beer
  :)

?

**Comment by svx** [comment]
I opened a new branch:

https://github.com/plone/documentation/tree/update_nginx and I also opened a new meta ticket for that: https://github.com/plone/documentation/issues/448

**Comment by tkimnguyen** [comment]
thx @svx - ok for me to add my changes to that new branch and close this PR?

**Comment by tkimnguyen** [comment]
ok @svx I hope this is what you wanted :) I added my nginx changes above to https://github.com/plone/documentation/tree/update_nginx and am closing this PR

**Pull request closed without merging**